### PR TITLE
Update UG and DG for recent changes

### LINKED ISSUE #92: Update user guide for refactored drive to pool

### LINKED ISSUE #115: Update DG to reflect change of targeted user
As user was changed from drivers/passengers to a HR executive that manages all of the records, the user cases and stories need to be updated.

### PR DESCRIPTION
**Pull request by lyueyang**
- As Program has been changed to accommodate a HR executive as the user instead of a passenger and driver, use case updated to reflect this change of focus.
- Full stops ended to each sentence
- MSS indicated

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -150,37 +150,36 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+|--------|--------------|-----------------------------------------------------------------|------------------------------------------------------------------------------------------------|
+|* * *   |HR Executive  |allocate drivers to passengers to be picked up                   |I can arrange carpooling trips for my colleagues                                                |
+|* * *   |HR Executive  |search for specific type of passengers                           |I can see if any passengers fulfil a criteria and view their carpool details                    |
+|* * *   |HR Executive  |list all passengers                                              |I can see all the passengers available                                                          |
+|* * *   |HR Executive  |create a profile                                                 |I can easily manage and track drivers and passengers                                            |
+|* * *   |HR Executive  |delete my profile                                                |passenger's data will not be stored when they are no longer looking to carpool                  |
```
Comment (ZechariahTan) [inline docs/DeveloperGuide.md]: I think you might have to 'change delete my profile to 'delete employee profile' as the user is the HR exec now and they're managing other employee's profiles.

Comment (lyueyang) [reply docs/DeveloperGuide.md]: Good catch, let me amend that

**Review by ZechariahTan — APPROVED** [review]
LGTM

**Review by ZechariahTan — APPROVED** [review]
LGTM

**Pull request merged**